File: Sources/SimERP.Data/DBEntities/Product.cs
using System;
using System.Collections.Generic;

namespace SimERP.Data.DBEntities
{
    public partial class Product
    {
        public int ProductId { get; set; }
        public string ProductCode { get; set; }
        public string Barcode { get; set; }
        public string Rfid { get; set; }
        public string ProductName { get; set; }
        public string ProductNameShort { get; set; }
        public decimal Price { get; set; }
        public decimal StandardCost { get; set; }
        public decimal PurchasePrice { get; set; }
        public Guid? ProductCategoryId { get; set; }
        public string ProductCategoryList { get; set; }
        public int UnitId { get; set; }
        public int TaxId { get; set; }
        public int? ImportTaxId { get; set; }
        public int ItemType { get; set; }
        public int ProductType { get; set; }
        public int CostMethod { get; set; }
        public int? SupplierId { get; set; }
        public string SupplierProductCode { get; set; }
        public string SupplierProductName { get; set; }
        public string SupplierNotes { get; set; }
        public string TermCondition { get; set; }
        public string ThumbnailPhoto { get; set; }
        public string LargePhoto { get; set; }
        public int? LeadTime { get; set; }
        public string MadeIn { get; set; }
        public int? PackageUnitId { get; set; }
        public decimal? PackageUnit { get; set; }
        public decimal? WeightUnit { get; set; }
        public bool IsUsingExpireDate { get; set; }
        public int? ExpireDays { get; set; }
        public bool IsItemVirtual { get; set; }
        public bool IsPackage { get; set; }
        public DateTimeOffset CreatedDate { get; set; }
        public int CreatedBy { get; set; }
        public DateTimeOffset? ModifyDate { get; set; }
        public int? ModifyBy { get; set; }
        public string SearchString { get; set; }
        public bool? IsActive { get; set; }
        public string Note { get; set; }
    }
}

File: Sources/SimERP/ClientApp/src/app/lists/product/model/product.ts
import {Attachfile} from './attachfile';
import {Guid} from 'guid-typescript';

export class Product {
  constructor() {

  }

  ProductId: number;
  ProductCode: string;
  Barcode: string;
  RFID: string;
  ProductName: string;
  ProductNameShort: string;
  Price: number;
  StandardCost: number;
  PurchasePrice: number;
  ProductCategoryId: Guid = undefined;
  ProductCategoryList: string;
  UnitId = -1;
  TaxId = -1;
  ImportTaxId = -1;
  ItemType = -1;
  ProductType = -1;
  CostMethod: number;
  SupplierId = -1;
  SupplierProductCode: string;
  SupplierProductName: string;
  SupplierNotes: string;
  TermCondition: string;
  ThumbnailPhoto: string;
  LargePhoto: string;
  LeadTime: number;
  MadeIn = '-1';
  PackageUnitId = -1;
  PackageUnit: number;
  WeightUnit: number;
  IsUsingExpireDate = false;
  ExpireDays: number;
  IsItemVirtual: boolean;
  IsPackage: boolean;
  CreatedDate: Date;
  CreatedBy: number;
  ModifyDate: Date;
  ModifyBy: number;
  SearchString: string;
  IsActive = false;
  Note: string;


  // custom properties
  UnitName: string;
  PackageUnitName: string;
  ProductCategoryName: string;
  CountryName: string;
  VendorName: string;
  ListAttachFile: Attachfile[] = [];
  ListAttachFileDelete: Attachfile[] = [];
}
